### DISCUSSION TITLE
Ultimaker and other CoreXY machine support

### DISCUSSION THREAD

**Discussion by Hello1024** [Sakati84/3DPrintColorizer#95] [Show and tell]
I have extended this project to the [Ultimaker 2](https://res.cloudinary.com/rsc/image/upload/b_rgb:FFFFFF,c_pad,dpr_1.0,f_auto,h_337,q_auto,w_600/c_pad,h_337,w_600/R8325379-07?pgw=1).   Since it's a Core-XY machine, a different mechanism is required for picking up pens.   I designed a mechanism to allow a pen to be collected and deposited using just X and Y motions.   With small code changes to the cura postprocessing plugin I got everything working nicely.

Is there interest in me publishing the details and a how-to guide?

**Comment by Nathan22211** [comment]
I would, just make sure the coloring works with your setup. I'd be willing to test it but my only CoreXY is being worked on and it's a mixing hotend one at that.

**Comment by StevenM2020** [comment]
I would be very interested in a how-to guide for a Core-xy.

**Comment by aamott** [comment]
I've been loosely keeping an eye on this project since it was started and even took a look at the code to see if I could add native CoreXY support but found myself deep in other projects. I ended up writing my own web version, a little more like a typical slicer's toolchangers: x, y, and z offset, start gcode, and end gcode. In case you never got your CoreXY working with this, here's mine: [3D Printer Marker Tracer](https://github.com/aamott/3D-Printer-Marker-Tracer)
